### PR TITLE
AppraiseFocus.lic

### PR DESCRIPTION
**Pull request by Windamere**
New script, Script does Appraise focus on Large edged weapon which can be added into yaml Hunting info - before - arg - appraisefocus. it Does the focus and then waits for Breakthrough before ending. If someone wants to rewrite it to make skill check appraise>200 or add arg's for different appraise focus items from the chart https://elanthipedia.play.net/Appraisal_skill#Compatible_Skills

Would also need gear.settings for different arg's

### CODE REVIEW

**Comment by Windamere** [comment]
Actually, only want to merge 1 file, the Sept. 2nd file....those from Aug 14th...not sure how to remove them out of my pull request..

File: AppraiseFocus.lic [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+match 'End', 'You have completed'
+fput('Appraise focus check')
+matchwait
+
+Wait:
+waitfor 'Breakthrough'
+echo "Time to buff up and go hunting!"
+
+End:
+fput(';k AppraiseFocus')
```
Comment (Windamere) [inline AppraiseFocus.lic]: This doesn't work, change to:   stop_script('appraisefocus')

**Comment by rpherbig** [comment]
I suggest rebasing onto master, there's current a conflict in `crossing-training`

**Comment by Windamere** [comment]
Sorry, No clue what that "rebasing onto master" means or how to do it...barely was able to get the pull request submitted and it has those extra files in it that i didn't even want in there lol....

> I suggest rebasing onto master, there's current a conflict in `crossing-training`

**Comment by jayRyan24** [comment]
Thank you for contributing changes Windamere! One thing I will say is that this isn't written in a way that makes sense with the rest of the code base. This is outdated in design and style, and if you want to include this into the main repo, it would need to be written in standard Ruby style. A guide for that can be found here: https://shopify.github.io/ruby-style-guide/ .  

Another thing is that this should instead of being a separate file, be brought into the appraisal code that already exists, as it's probably a good inclusion. A bad thing within coding is something called hard-coded settings. This, for instance in your code, would be the broadsword. Not everyone has one. With Dependency, being a shared code base, it would need to use the YAML settings (or an argument) for such an item. If you need help with this, I am sure more than enough people would be willing to help over lnet and drscripts. 

If you aren't up for those things, this is a good start and could be made into an Issue that you could file and ask someone to implement it into the master code base in a more standard way.

Thanks again.

File: AppraiseFocus.lic [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+match 'End', 'You have completed'
+fput('Appraise focus check')
+matchwait
+
+Wait:
+waitfor 'Breakthrough'
+echo "Time to buff up and go hunting!"
+
+End:
+kill_script('AppraiseFocus')
```
Comment (jayRyan24) [inline AppraiseFocus.lic]: This just needs to be exit.

File: AppraiseFocus.lic [hunk 3]
```diff
@@ -0,0 +1,35 @@
+waitrt?
+match 'Focus', "You get"
+match 'Focus', "You are already holding"
+fput('get my broadsword')
```
Comment (jayRyan24) [inline AppraiseFocus.lic]: broadsword needs to be a setting brought into this and not hard coded.

**Comment by jayRyan24** [comment]
> Sorry, No clue what that "rebasing onto master" means or how to do it...barely was able to get the pull request submitted and it has those extra files in it that i didn't even want in there lol....
> 
> > I suggest rebasing onto master, there's current a conflict in `crossing-training`

It just means your master branch is outdated, meaning that changes have been made since your last update and you need to pull them into your code. They're basically uneven. 

https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

**Comment by Windamere** [comment]
> Thank you for contributing changes Windamere! One thing I will say is that this isn't written in a way that makes sense with the rest of the code base. This is outdated in design and style, and if you want to include this into the main repo, it would need to be written in standard Ruby style. A guide for that can be found here: https://shopify.github.io/ruby-style-guide/ .
> 
> Another thing is that this should instead of being a separate file, be brought into the appraisal code that already exists, as it's probably a good inclusion. A bad thing within coding is something called hard-coded settings. This, for instance in your code, would be the broadsword. Not everyone has one. With Dependency, being a shared code base, it would need to use the YAML settings (or an argument) for such an item. If you need help with this, I am sure more than enough people would be willing to help over lnet and drscripts.
> 
> If you aren't up for those things, this is a good start and could be made into an Issue that you could file and ask someone to implement it into the master code base in a more standard way.
> 
> Thanks again.

Yes I would very much like for anyone, that wants, to rewrite it. My scripting knowledge is very limited and did what i could using what i know...Thanks i'll check out that Ruby link..

**Comment by jayRyan24** [comment]
@Windamere Can you make this into an issue and then close this PR. I will implement this later today into the appraisal script. I will make it work with `;appraisal focus <item>`.

**Comment by jayRyan24** [comment]
As a side note, how does this train? When does focus appraisal begin to train, and how high does it train to? Does it only work with weapons? Etc., thanks.

**Comment by Windamere** [comment]
> As a side note, how does this train? When does focus appraisal begin to train, and how high does it train to? Does it only work with weapons? Etc., thanks.

ok it takes 200 ranks of appraisal (or clockwork shark item). It doesn't train per se itself meaning it doesn't train appraise at all, it gives bonus' to absoption while training the item you focus. To train it, You appraise focus item then wait until breakthrough just like research. It gives different bonus' based on the item. example Appraise focus Broadsword give bonus to large edge absorption when using, But Appraise focus box give bonus's to locksmithing absorption.  This bonus does NOT stack with an RPA bonus.. but is slightly better than no bonus when not using an RPA award.

**Pull request closed without merging**